File: P5 2024-2/Paradigmas de Programacao/ListaExercicio01/Golang/Configuracao.go
package main

import (
	"sync"
)

type Configuracao struct {
	valor string
}

var instancia *Configuracao
var once sync.Once

func ObterConfiguracao() *Configuracao {
	once.Do(func() {
		instancia = &Configuracao{valor: "Configuração padrão"}
	})
	return instancia
}

File: P5 2024-2/Paradigmas de Programacao/ListaExercicio01/Python/Configuracao.py
class Configuracao:
    _instancia = None
    def __init__(self, valor = "Configuracao Padrao"):
        self.valor = valor

    def __new__(config):
        if config._instancia is None:
            config._instancia = super(Configuracao, config).__new__(config)
        return config._instancia
